app/models/conversation.rb

```diff
@@ -58,6 +58,6 @@ def self.grouped_by_increasing_time_interval_for_user(user)
   private
 
   def set_title_async
-    AutotitleConversationJob.perform_later(id)
+    AutotitleConversationJob.set(wait: 2.seconds).perform_later(id) # it's more important to let the queue prioritize the message response so we delay this a bit
   end
```

Comment:
Further up in this class, the trigger is `after_create_commit` if `title.blank?` . I struggled with the proper trigger because Message.create creates a conversation a split second before a Message is created, so if the job runs too fast it could run before the message has been created.

Initially, I tried changing this to after_update_commit since the creation of a message updates the conversation timestamp and then I added messages.count >= 1. But we update conversation A TON because every single update to a message fragment while it's streaming in and so it was creating a ton of duplicate set_title jobs. (I actually wonder if we need to make the quantity of messages updates more efficient?)

So I changed this back to after_create_commit and I added a 2 second delay (down in set_title_async).

But... there's a possibility it runs once and still fails to find any messages. I just added a retry mechanism to help with that.